### PR TITLE
[freetype] Disabled Harfbuzz check for non-Windows platforms

### PR DESCRIPTION
**Pull request by capnkenny**
- What does your PR fix? 
Fixes issue #10041 
Resolves linking errors when building on Linux-based platforms

- Which triplets are supported/not supported? Have you updated the CI baseline?
 No change.
 
- Does your PR follow the maintainer guide?
 I think so.

### CODE REVIEW

File: ports/freetype/portfile.cmake [hunk 1]
```diff
@@ -31,15 +31,28 @@ else()
...
+	vcpkg_configure_cmake(
+		SOURCE_PATH ${SOURCE_PATH}
+		PREFER_NINJA
+		OPTIONS
+			-DCONFIG_INSTALL_PATH=share/freetype
+			-DFT_WITH_ZLIB=ON # Force system zlib.
+			${FEATURE_OPTIONS}
+			-DENABLE_DLL_EXPORT=${ENABLE_DLL_EXPORT}
+	)
+endif()
```
Comment (PhoebeHui) [inline ports/freetype/portfile.cmake]: Could you please update this to

```
set(OPTIONS)
if (NOT VCPKG_TARGET_IS_WINDOWS)
  list(APPEND OPTIONS -DFT_DISABLE_FIND_PACKAGE_Harfbuzz=TRUE)
endif()

vcpkg_configure_cmake(
    SOURCE_PATH ${SOURCE_PATH}
    PREFER_NINJA
    OPTIONS
        -DCONFIG_INSTALL_PATH=share/freetype
        -DFT_WITH_ZLIB=ON # Force system zlib.
        ${FEATURE_OPTIONS}
        -DENABLE_DLL_EXPORT=${ENABLE_DLL_EXPORT}
        ${OPTIONS}
)
```

**Review by PhoebeHui — CHANGES_REQUESTED** [review]
@capnkenny, thanks for the PR!

Could you also update the version to '2.10.1-5' in [CONTROL](https://github.com/microsoft/vcpkg/blob/5f12ffae8d6444383ed6d88da08ee32876726ebc/ports/freetype/CONTROL) file?

File: ports/freetype/portfile.cmake (hunk 1 above)
Comment (capnkenny) [reply ports/freetype/portfile.cmake]: Made both changes as requested 👍

**Comment by PhoebeHui** [comment]
The failures doesn't related to this changes, I will help  rerun after the baseline issue fixed.

**Review by PhoebeHui — APPROVED** [review]

**Comment by strega-nil** [comment]
Cool, thanks @capnkenny :)

**Pull request merged**